### PR TITLE
Extract `NumericData` model for tests

### PR DESCRIPTION
**Pull request by kamipo**
Currently `NumericData` model is defined some places.

### CODE REVIEW

File: activerecord/test/cases/base_test.rb [hunk 1]
```diff
@@ -908,13 +909,6 @@ def test_default
     end
   end
 
-  class NumericData < ActiveRecord::Base
-    self.table_name = "numeric_data"
-
-    attribute :my_house_population, :integer
-    attribute :atoms_in_universe, :integer
-  end
-
```
Comment (prathamesh-sonpatki) [inline activerecord/test/cases/base_test.rb]: Is this not needed now?

File: activerecord/test/cases/calculations_test.rb [hunk 2]
```diff
@@ -17,14 +18,6 @@
 require "models/rating"
 require "models/post"
 
-class NumericData < ActiveRecord::Base
-  self.table_name = "numeric_data"
-
-  attribute :world_population, :integer
-  attribute :my_house_population, :integer
-  attribute :atoms_in_universe, :integer
-end
```
Comment (prathamesh-sonpatki) [inline activerecord/test/cases/calculations_test.rb]: Is this also not needed now?

Comment (kamipo) [inline activerecord/test/cases/calculations_test.rb]: Yes, this was added for prevent deprecation message in #15295.
But the deprecation message was removed in #15374.
We do not see deprecation message even if these line are removed.

**Comment by rafaelfranca** [comment]
Related with https://github.com/rails/rails/commit/aff73e094640b19cb1bdc975082be7b4767bf4bf. We should put the deprecation back what will require the override of the attributes in the tests.

**Comment by kamipo** [comment]
Restored the override of the attributes properly.

**Pull request merged**